skbio/diversity/alpha/_base.py

```diff
@@ -848,6 +850,51 @@ def osd(counts):
     return observed_otus(counts), singles(counts), doubles(counts)
 
 
+@experimental(as_of="0.4.0-dev")
+def pielou_e(counts):
+    r"""Calculate Pielou's Evenness index J'.
+
+    Pielou's Evenness is defined as:
+
+    .. math::
+
+       J' = \frac{(H)}{\ln(S)}
+
+    where :math:`H` is the Shannon-Wiener entropy of counts and :math:`S` is
+    the number of OTUs in the sample.
+
+    Parameters
+    ----------
+    counts : 1-D array_like, int
+        Vector of counts.
+
+    Returns
+    -------
+    double
+        Pielou's Evenness.
+
+    See Also
+    --------
+    shannon
+    heip_e
+
+    Notes
+    -----
+    The implementation here is based on the description given by Carlo Heip
+    [1]_ and is also described on Wikipedia [2]_. It is similar to Heip's
+    evenness [1]_.
+
+    References
+    ----------
+    .. [1] Heip, C. 1974. A new index measuring evenness. J. Mar. Biol. Ass.
+       UK., 54, 555-557.
+    .. [2] https://en.wikipedia.org/wiki/Species_evenness
+
+    """
+    counts = _validate_counts_vector(counts)
+    return shannon(counts) / np.log(observed_otus(counts))
```

Comment:
Thanks for the details @wdwvt1! I agree that changing the default base to `e` is probably safest for consistency, though outside this PR's scope.

@ebolyen and I discussed these border cases, let's go with the following for now:

- for empty input `[]`, let it continue to raise an error (don't worry about adding unit tests for this case). The current error message isn't that helpful but we will revisit how to handle `[]` universally when we bring the alpha diversity package from experimental to stable (timeline TBD).

- for counts of all zeros (e.g., `[0, 0, 0]`) add a unit test ensuring it returns `nan`

- for a single species occurrence (e.g., `[0, 42, 0]` and `[42]`) add a unit test ensuring it returns `nan`. It is unclear whether an evenness of 1.0 should be returned, so for now we'll match the equation and return `nan`. We'll revisit this decision in the future when working on the alpha diversity package.